#### CS 457 - Homework Assignment 4: Statistical Inference
##### Ali Hashir - ah05433@st.habib.edu.pk

In [34]:
import pandas as pd
import numpy as np
from scipy import stats


In [35]:
df = pd.read_csv('EmployeeAttrition.csv')
df.head()

Age     BusinessTravel  DailyRate              Department   
0   41      Travel_Rarely       1102                   Sales  \
1   49  Travel_Frequently        279  Research & Development   
2   37      Travel_Rarely       1373  Research & Development   
3   33  Travel_Frequently       1392  Research & Development   
4   27      Travel_Rarely        591  Research & Development   

   DistanceFromHome  Education EducationField  EnvironmentSatisfaction   
0                 1          2  Life Sciences                        2  \
1                 8          1  Life Sciences                        3   
2                 2          2          Other                        4   
3                 3          4  Life Sciences                        4   
4                 2          1        Medical                        1   

   Gender  HourlyRate  ...  StandardHours  StockOptionLevel TotalWorkingYears   
0  Female          94  ...             80                 0                 8  \
1    Male          61  ...             80                 1                10   
2    Male          92  ...             80                 0                 7   
3  Female          56  ...             80                 0                 8   
4    Male          40  ...             80                 1                 6   

   TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  YearsInCurrentRole   
0                      0               1               6                   4  \
1                      3               3              10                   7   
2                      3               3               0                   0   
3                      3               3               8                   7   
4                      3               3               2                   2   

   YearsSinceLastPromotion YearsWithCurrManager  Attrition  
0                        0                    5        Yes  
1                        1                    7         No  
2                        0                    0        Yes  
3                        3                    0         No  
4                        2                    2         No  

[5 rows x 32 columns]

---

#### Q1. If the MonthlyIncome of Males is greater than Females
H0 : Monthly income of males is almost equal to the monthly income of females. </br>
HA : Monthly income of males is greater than monthly income of females.


In [36]:
male = df[(df['Gender']=='Male')]['MonthlyIncome']
female = df[(df['Gender']=='Female')]['MonthlyIncome']
stats.ttest_ind(male, female, alternative='greater')

Ttest_indResult(statistic=-1.2212617308870655, pvalue=0.8889084827245605)

##### Conclusion
Since the p-value > 0.05 (Not rejectiong null hypothesis), the monthly income of males is almost equal to the monthly income of females. This suggests that the company does not exhibit gender bias when it comes to monthly income, indicating a fair compensation policy regardless of gender. (p-value: 0.8889)

---

#### Q2. If the WorkLifeBalance of Males is less than Females
H0 : The WorkLifeBalance of Males is almost equal than that of Females </br>
HA : The WorkLifeBalance of Males is less than that of Females


In [37]:
male = df[(df['Gender'] == 'Male')]['WorkLifeBalance']
female = df[(df['Gender'] == 'Female')]['WorkLifeBalance']
stats.ttest_ind(male, female, alternative='less')


Ttest_indResult(statistic=-0.10546787956064164, pvalue=0.45800948246831286)

##### Conclusion
Since the p-value > 0.05 (not rejecting null hypothesis), the WorkLifeBalance of males is almost equal to that of females. This highlights that the company promotes a balanced work-life environment for both genders. (p-value: 0.4580)

---

#### Q3. If the YearsAtCompany of Single is less than Married
H0 : The YearsAtCompany of Single is almost equal than that of Married </br>
HA : The YearsAtCompany of Single is less than Married


In [38]:
single = df[(df['MaritalStatus'] == 'Single')]['YearsAtCompany']
married = df[(df['MaritalStatus'] == 'Married')]['YearsAtCompany']
stats.ttest_ind(single, married, alternative='less')


Ttest_indResult(statistic=-2.5589985737464476, pvalue=0.0053127663549315815)

##### Conclusion
Since the p-value < 0.05 (rejecting null hypothesis and accepting alternate hypotheses), the YearsAtCompany of single individuals is less than that of married individuals. This might suggest that married employees tend to stay longer with the company, possibly due to factors like stability or benefits that the company offers or they might need extra pay. (p-value: 0.0053)

---

#### Q4. If the EnvironmentalSatisfaction of Attrition=Yes is less than Attrition=No
H0 : The EnvironmentalSatisfaction of Attrition=Yes is almost equal than Attrition=No </br>
HA : The EnvironmentalSatisfaction of Attrition=Yes is less than Attrition=No


In [39]:
yes = df[(df['Attrition'] == 'Yes')]['EnvironmentSatisfaction']
no = df[(df['Attrition'] == 'No')]['EnvironmentSatisfaction']
stats.ttest_ind(yes, no, alternative='less')


Ttest_indResult(statistic=-3.9818600026818496, pvalue=3.58616927468443e-05)

##### Conclusion
Since the p-value < 0.05 (rejecting null hypothesis and accepting alternate hypotheses), the EnvironmentalSatisfaction of employees with Attrition=Yes is less than those with Attrition=No. This indicates that the work environment plays a significant role in employee retention. (p-value: 3.58616927468443e-05)

---

#### Q5. If the MonthlyIncome of Manager is greater than Laboratory Technician (Hint: Use JobRole to find Manager and Laboratory Technician)
H0 : The MonthlyIncome of Manager is almost equal than Laboratory Technician </br>
HA : The MonthlyIncome of Manager is greater than Laboratory Technician


In [40]:
manager = df[(df['JobRole'] == 'Manager')]['MonthlyIncome']
laboraty_technician = df[(df['JobRole'] == 'Laboratory Technician')]['MonthlyIncome']
stats.ttest_ind(manager, laboraty_technician, alternative='greater')


Ttest_indResult(statistic=76.03317414634023, pvalue=9.971168334368433e-224)

##### Conclusion
Since the p-value < 0.05 (rejecting null hypothesis and accepting alternate hypotheses), the MonthlyIncome of a manager is greater than that of a laboratory technician. This is expected, as managerial roles often come with higher compensation reflecting the difference in responsibilities and skill requirements. (p-value: 9.971168334368433e-224)

---

#### Q6. Is there any association between the Department and Gender? 
H0 : Department and gender are independant </br>
HA : Department and gender are independent.


In [43]:
chisqt = pd.crosstab(df['Department'],df['Gender'], margins=True)
value = np.array([chisqt.iloc[0][0:].values,chisqt.iloc[1][0:].values])
a, p, b, c = stats.chi2_contingency(value)
p


0.47927963769548776

##### Conclusion
Since the p-value > 0.05 (not rejectiong null hypothesis), department and gender are independent and have no association. This shows that the company's departmental distribution does not show any gender preference, indicating a gender-neutral hiring process. (p-value: 0.4792)

---